internal/querynodev2/delegator/delegator_data.go

```diff
@@ -377,11 +377,13 @@ func (sd *shardDelegator) LoadSegments(ctx context.Context, req *querypb.LoadSeg
 		}
 	})
 	log.Info("load delete...")
-	err = sd.loadStreamDelete(ctx, candidates, infos, req.GetDeltaPositions(), targetNodeID, worker, entries)
+	err = sd.loadStreamDelete(ctx, candidates, infos, req.GetDeltaPositions(), targetNodeID, worker)
 	if err != nil {
 		log.Warn("load stream delete failed", zap.Error(err))
 		return err
 	}
+	// alter distribution
+	sd.distribution.AddDistributions(entries...)
```

Comment:
altering distribution shall be protected by delete lock otherwise some delete entry might be missed